### PR TITLE
Add deployed contracts page in teams layout

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to add a progress component, update styles, improve UI elements, and enhance navigation in the dashboard app.

### Detailed summary
- Added a new `Progress` component
- Updated styles for `ScrollShadow` and `NetworkSelectDropdown`
- Improved UI elements like badges and modals
- Enhanced navigation with new layout components

> The following files were skipped due to too many changes: `apps/dashboard/src/components/dashboard/StepsCard.stories.tsx`, `apps/dashboard/src/@/components/ui/select.tsx`, `apps/dashboard/src/app/team/[team_slug]/[project_slug]/contracts/_components/GetStartedWithContractsDeploy.tsx`, `apps/dashboard/src/components/dashboard/StepsCard.tsx`, `apps/dashboard/src/pages/dashboard/contracts/deploy.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
* **#4440** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4440?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/@/components/blocks/MobileSidebar.tsx [hunk 1]
```diff
@@ -15,8 +15,9 @@ export function MobileSidebar(props: {
     <Dialog open={isOpen} onOpenChange={setIsOpen}>
       <DialogTrigger asChild>{props.trigger}</DialogTrigger>
       <DialogContent
-        className="p-4 rounded-t-xl rounded-b-none"
+        className="p-4 rounded-t-xl rounded-b-none z-[9001]"
```
Comment (jnsdls) [inline apps/dashboard/src/@/components/blocks/MobileSidebar.tsx:18]: this scares me, why do we need the custom z-index?

Comment (MananTank) [reply apps/dashboard/src/@/components/blocks/MobileSidebar.tsx]: good catch, this was a temp fix. Added proper fix and removed it

**Review by jnsdls — CHANGES_REQUESTED** [review]
the scroll behavior on this feels super janky, stutters etc, can we just have it be normal overflow scroll?

**Comment by MananTank** [comment]
Removed this functionality from select component

**Review by jnsdls — APPROVED** [review]

**Pull request merged**